pkg/ds/daemon_set.go

```diff
@@ -392,6 +413,111 @@ func (ds *daemonSet) unschedule(node types.NodeName) error {
 	return nil
 }
 
+func (ds *daemonSet) PublishToReplication() error {
+	// TODO: We need to specifically turn this on to work
+	if ds.healthChecker == nil {
+		ds.logger.Info("Healthchecker is nil")
+		return nil
+	}
+
+	podLocations, err := ds.CurrentPods()
+	if err != nil {
+		return util.Errorf("Error retrieving pod locations from daemon set: %v", err)
+	}
+	nodes := podLocations.Nodes()
+
+	ds.logger.Infof("Preparing to publish the following nodes: %v", nodes)
+
+	thisHost, err := os.Hostname()
+	if err != nil {
+		log.Fatalf("Could not retrieve hostname: %s", err)
+	}
+	thisUser, err := user.Current()
+	if err != nil {
+		log.Fatalf("Could not retrieve user: %s", err)
+	}
+	lockMessage := fmt.Sprintf("%q from %q at %q", thisUser.Username, thisHost, time.Now())
+	repl, err := replication.NewReplicator(
+		ds.DaemonSet.Manifest,
+		ds.logger,
+		nodes,
+		len(nodes)-ds.DaemonSet.MinHealth,
+		ds.kpStore,
+		ds.applicator,
+		*ds.healthChecker,
+		health.HealthState("warning"),
+		lockMessage,
+	)
+	if err != nil {
+		for _, node := range nodes {
+			ds.unschedule(node)
```

Comment:
I don't think you want to wipe out all of the pods if the replication fails, just bail out